mirrord/cli/src/main.rs

```diff
@@ -124,21 +131,49 @@ where
     );
     sub_progress_config.success(Some("config summary"));
 
+    // since execvpe doesn't exist on macOS, resolve path manually and use execve
+    let binary_path: String = if let Some(path) = var_os("PATH") {
+        let mut resolved = binary.clone();
+        let paths = split_paths(&path).collect::<Vec<_>>();
+        for mut path in paths {
+            // check if binary is present in each path
+            path.push(binary.clone());
+            if path.is_executable() {
+                resolved = path.to_string_lossy().into();
```

Comment:
I think we could avoid the lossy conversion, if you want

```suggestion
                resolved = CString::new(path.as_ref().as_os_str().as_bytes())?;
```